### PR TITLE
Change StreamletDescriptor, remove image and projectId - MultiImage epic

### PR DESCRIPTION
**Pull request by debasishg**
### What changes were proposed in this pull request?

* This is one of the tasks in multi-image epic where the user would like to deploy a cloudflow application with multiple images. This one refactors `ApplicationDescriptor` and associated changes in `StreamletDescriptor`, `StreamletDeployment` and other classes.
* Associated changes in CLI

### Why are the changes needed?

* This is one of the first steps towards multi-image cloudflow application deployment 

### Does this PR introduce any user-facing change?

* Yes, it has user facing changes as it's an architectural change for cloudflow

### How was this patch tested?

* All unit tests are made to run green
* Tested the following configurations on GKE (no changes in operator):
  * New library + new CLI => works
  * Old library + new CLI   => works
  * New library + old CLI   => works

### CODE REVIEW

File: core/cloudflow-blueprint/src/main/scala/cloudflow/blueprint/deployment/ApplicationDescriptor.scala [hunk 1]
```diff
@@ -54,6 +54,7 @@ object ApplicationDescriptor {
   def apply(
       appId: String,
       appVersion: String,
+      img: String,
```
Comment (RayRoestenburg) [inline core/cloudflow-blueprint/src/main/scala/cloudflow/blueprint/deployment/ApplicationDescriptor.scala]: very minor `img` -> `image`?

Comment (debasishg) [reply core/cloudflow-blueprint/src/main/scala/cloudflow/blueprint/deployment/ApplicationDescriptor.scala]: @RayRoestenburg What is the Scala syntax of a case class copy when both the names are same ? Will `deployment.copy(image = image)` work ?

Comment (RayRoestenburg) [reply core/cloudflow-blueprint/src/main/scala/cloudflow/blueprint/deployment/ApplicationDescriptor.scala]: yes.

Comment (debasishg) [reply core/cloudflow-blueprint/src/main/scala/cloudflow/blueprint/deployment/ApplicationDescriptor.scala]: done ..

File: core/cloudflow-blueprint/src/main/scala/cloudflow/blueprint/deployment/ApplicationDescriptor.scala [hunk 2]
```diff
@@ -54,6 +54,7 @@ object ApplicationDescriptor {
   def apply(
       appId: String,
       appVersion: String,
+      image: String,
```
Comment (agemooij) [inline core/cloudflow-blueprint/src/main/scala/cloudflow/blueprint/deployment/ApplicationDescriptor.scala]: Shouldn't this (eventually) be a list of images?

**Comment by agemooij** [comment]
@debasishg great wrk on the backwards compatibility!

**Pull request merged**